### PR TITLE
Logging: Fix don't return an error trying to initialize the file logger

### LINKED ISSUE #42897: 8.2.x ->8.3.1 Upgrade
**What happened**:
The normal DEB update release will no longer run under a grafana user/group without throwing errors related to both the log folder and /var/run/grafana. The recent release upgrade broke a system that i've upgraded without issue since 6.x and even uninstalling and reinstalling (checking all paths are removed) did not resolve the issue. The disk is NOT running in RO mode, I've manually confirmed this by touching another file in the same path.

Dec 08 13:11:19 host01 grafana-server[2560]: Failed to start grafana. error: failed to initialize file handler: open /usr/share/grafana/data/log/grafana.log: read-only file system
Dec 08 13:11:19 host01 grafana-server[2560]: failed to initialize file handler: open /usr/share/grafana/data/log/grafana.log: read-only file system

**What you expected to happen**:
The upgrade and related permission requirements would be applied or at least known to change.

**How to reproduce it (as minimally and precisely as possible)**:
Install 8.3.1 from apt deb package.

**Anything else we need to know?**:
I had this issue after the upgrade and after purging all grafana packages and manually confirming and removing the remaining folders it came back with a few install.

The grafana user/group has /bin/false set for the login shell.

**Environment**:
- Grafana version: 8.3.1
- Data source type & version: Postgres
- OS Grafana is installed on: Ubuntu 18.04LTS
- User OS & Browser: n/a
- Grafana plugins: n/a
- Others: n/a

### PR DESCRIPTION
**Pull request by bolshoytoster**
**What this PR does / why we need it**:
Before, when grafana fails to initialize file logging it would completely abort.

This short commit makes it so that grafana will simply skip initializing file logging.

**Which issue(s) this PR fixes**:

Fixes #42897

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/infra/log/log.go [hunk 1]
```diff
@@ -161,7 +161,9 @@ func ReadLoggingConfig(modes []string, logsPath string, cfg *ini.File) error {
 			fileHandler.Maxdays = sec.Key("max_days").MustInt64(7)
 			if err := fileHandler.Init(); err != nil {
 				Root.Error("Failed to initialize file handler", "dpath", dpath, "err", err)
-				return errutil.Wrapf(err, "failed to initialize file handler")
+				// uncomment to crash when it can't create a log file
```
Comment (ying-jeanne) [inline pkg/infra/log/log.go:164]: we probably want to do the same on few lines above when create the directory
```
			if err := os.MkdirAll(dpath, os.ModePerm); err != nil {
				Root.Error("Failed to create directory", "dpath", dpath, "err", err)
				return errutil.Wrapf(err, "failed to create log directory %q", dpath)
			}
```
to continue instead returning error to keep the behavior consistent

Comment (bolshoytoster) [reply pkg/infra/log/log.go:166]: I've added this.

**Comment by marefr** [comment]
@bolshoytoster can you please sync/merge with latest main branch changes for us to be able to review this properly. Thanks

**Review by marefr — APPROVED** [review]
LGTM

**Comment by marefr** [comment]
@bolshoytoster can you please merge with main again since there a conflict?

**Review by ying-jeanne — APPROVED** [review]

File: pkg/infra/log/log.go [hunk 2]
```diff
@@ -335,8 +335,9 @@ func ReadLoggingConfig(modes []string, logsPath string, cfg *ini.File) error {
 			fileHandler.Daily = sec.Key("daily_rotate").MustBool(true)
 			fileHandler.Maxdays = sec.Key("max_days").MustInt64(7)
 			if err := fileHandler.Init(); err != nil {
+				continue
```
Comment (marefr) [inline pkg/infra/log/log.go:338]:
```suggestion
```

**Comment by marefr** [comment]
@bolshoytoster I merged with main and pushed the changes.

**Comment by marefr** [comment]
Thank you for contributing to Grafana

**Pull request merged**